tools/structfield/testdata/src/has-warnings/main.go

```diff
@@ -5,18 +5,33 @@
-	DollarRef        string  `json:"$ref"`                       // want `change Go field name "DollarRef" to "Ref" for tag "\$ref" in struct "Example"`
+type JSONFieldName struct {
+	GitHubThing      string  `json:"github_thing"`               // want `change Go field name "GitHubThing" to "GithubThing" for tag "github_thing" in struct "JSONFieldName"`
+	Id               *string `json:"id,omitempty"`               // want `change Go field name "Id" to "ID" for tag "id" in struct "JSONFieldName"`
+	strings          *string `json:"strings,omitempty"`          // want `change Go field name "strings" to "Strings" for tag "strings" in struct "JSONFieldName"`
+	camelcaseexample *int    `json:"camelCaseExample,omitempty"` // want `change Go field name "camelcaseexample" to "CamelCaseExample" for tag "camelCaseExample" in struct "JSONFieldName"`
+	DollarRef        string  `json:"$ref"`                       // want `change Go field name "DollarRef" to "Ref" for tag "\$ref" in struct "JSONFieldName"`
 }
 
 type JSONFieldType struct {
-	ID string `json:"id,omitempty"` // want `change the \"ID\" field type to "\*string" in the struct "JSONFieldType" because its tag uses "omitempty"`
+	String                         string             `json:"string,omitempty"`                              // want `change the "String" field type to "\*string" in the struct "JSONFieldType" because its tag uses "omitempty"`
+	SliceOfStringPointers          []*string          `json:"slice_of_string_pointers,omitempty"`            // want `change the "SliceOfStringPointers" field type to "\[\]string" in the struct "JSONFieldType"`
+	PointerToSliceOfStrings        *[]string          `json:"pointer_to_slice_of_strings,omitempty"`         // want `change the "PointerToSliceOfStrings" field type to "\[\]string" in the struct "JSONFieldType"`
+	SliceOfStructs                 []Struct           `json:"slice_of_structs,omitempty"`                    // want `change the "SliceOfStructs" field type to "\[\]\*Struct" in the struct "JSONFieldType"`
+	PointerToSliceOfStructs        *[]Struct          `json:"pointer_to_slice_of_structs,omitempty"`         // want `change the "PointerToSliceOfStructs" field type to "\[\]\*Struct" in the struct "JSONFieldType"`
+	PointerToSliceOfPointerStructs *[]*Struct         `json:"pointer_to_slice_of_pointer_structs,omitempty"` // want `change the "PointerToSliceOfPointerStructs" field type to "\[\]\*Struct" in the struct "JSONFieldType"`
+	PointerToMap                   *map[string]string `json:"pointer_to_map,omitempty"`                      // want `change the "PointerToMap" field type to "map\[string\]string" in the struct "JSONFieldType"`
+	SliceOfInts                    []*int             `json:"slice_of_ints,omitempty"`                       // want `change the "SliceOfInts" field type to "\[\]int" in the struct "JSONFieldType"`
```

Comment:
Could you please also demonstrate that `[]*int64` and `[]*bool` and `[]*float64` also get flagged properly since those are probably going to be the most common after `int`?

Reply:
OK, please ignore this. LGTM. Merging.